std/experimental/allocator/building_blocks/free_list.d

```diff
@@ -1045,6 +1047,28 @@ struct SharedFreeList(ParentAllocator,
         resetNodes();
         return result;
     }
+
+    /**
+    Nonstandard function that minimizes the memory usage of the freelist by
+    freeing each element in turn. Defined only if $(D ParentAllocator) defines
+    $(D deallocate).
+    */
+    static if (hasMember!(ParentAllocator, "deallocate") && !unchecked)
+    void minimize() shared
+    {
+        lock.lock();
+        scope(exit) lock.unlock();
```

Comment:
A bit uncomfortable about a potentially long loop in a critical section. One alternative would be to lock upon each deallocation. But then the function is liable to run for a long time, too, if many other threads create free nodes. Another would be to lock every N deallocations. And yet another would be to run at most N times. I guess the proposed implementation is a good start.